inst/assumptions.m

```diff
@@ -111,10 +111,9 @@
       'd = x._assumptions.generator'
       'if d == {}:'
       '    astr = ""'
-      'elif all(d.values()):'  % all True so list them
-      '    astr = ", ".join(sorted([str(i) for i in d.keys()]))'
-      'else:'  % more complicated case, just the raw dict
-      '    astr = str(d)'
+      'else:'  % all True so list them
+      '    val, key = list(d.values()), list(d.keys())' % In Python3 d.values() don't support indexing
+      '    astr = ", ".join(sorted([(("" if val[i] else "~") + str(key[i])) for i in range(len(d.keys()))]))'
```

Comment:
Yes, theoretically only can be true or false, and that says the doc, i test this with none and only causes irregular things and bugs...., so i don't think we will get that value in the future.